### PR TITLE
forEach statement + tests

### PR DESCRIPTION
**Pull request by dafik**

### CODE REVIEW

**Comment by c9s** [comment]
Can you use rebase to avoid the unintended merge point? Thanks

File: src/CodeGen/Statement/ForEachStatement.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            $this->forEachBlock = new Block();
+        }
+    }
+
+    public function render(array $args = array())
+    {
+        $this->forEachBlock->setIndentLevel($this->indentLevel + 1);
+
+        if ($this->keyVariable) {
+            $this->lines[] = 'foreach(' . $this->forVariable . ' as ' . $this->keyVariable . ' => ' . $this->asVariable . '){';
```
Comment (c9s) [inline src/CodeGen/Statement/ForEachStatement.php]: should be `foreach (`

Comment (c9s) [inline src/CodeGen/Statement/ForEachStatement.php]: And `) {`

File: src/CodeGen/Statement/ForEachStatement.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        } else {
+            $this->lines[] = 'foreach(' . $this->forVariable . ' as ' . $this->asVariable . '){';
```
Comment (c9s) [inline src/CodeGen/Statement/ForEachStatement.php]: also here `foreach (` and `) {`

**Comment by dafik** [comment]
I'm not very familiar with git, I'll try to learn how to use rebase, maybe You know a good explanation how to use it ?

**Comment by c9s** [comment]
I usually use

git pull --rebase upstream master

upstream : your upstream remote name
master: the remote branch that you want to pull down

dafik notifications@github.com 於 2015年11月10日 星期二寫道：

**Comment by dafik** [comment]
spaces added

**Comment by c9s** [comment]
And 'git rebase' will rebase your current branch. You will need to force push your changes to your repo

**Comment by c9s** [comment]
I've also rebased the master branch

**Pull request merged**